cli/file_fetcher.rs

```diff
@@ -2,677 +2,499 @@
-        | "text/javascript"
-        | "application/ecmascript"
-        | "text/ecmascript"
-        | "application/x-javascript"
-        | "application/node" => {
-          map_js_like_extension(path, MediaType::JavaScript)
-        }
-        "application/json" | "text/json" => MediaType::Json,
-        "application/wasm" => MediaType::Wasm,
-        // Handle plain and possibly webassembly
-        "text/plain" | "application/octet-stream" => MediaType::from(path),
-        _ => {
-          debug!("unknown content type: {}", content_type);
-          MediaType::Unknown
-        }
+  /// Fetch a source file and asynchronously return it.
+  pub async fn fetch(
+    &self,
+    specifier: &ModuleSpecifier,
+    permissions: &Permissions,
+  ) -> Result<File, AnyError> {
+    debug!("FileFetcher::fetch() - specifier: {}", specifier);
+    let scheme = get_scheme(specifier)?;
+    permissions.check_specifier(specifier)?;
+    if let Some(file) = self.cache.get(specifier) {
+      debug!("  found in in-memory cache");
+      Ok(file)
+    } else {
+      let is_local = scheme == "file";
+
+      let result = if is_local {
+        fetch_local(specifier)
+      } else if !self.allow_remote {
+        Err(custom_error(
+          "NoRemote",
+          format!("A remote specifier was requested: \"{}\", but --no-remote is specified.", specifier),
+        ))
+      } else {
+        self.fetch_remote(specifier, permissions, 10).await
       };
 
-      let charset = ct_iter
-        .map(str::trim)
-        .find_map(|s| s.strip_prefix("charset="))
-        .map(String::from);
+      if let Ok(file) = &result {
+        self.cache.insert(specifier.clone(), file.clone());
+      }
 
-      (media_type, charset)
+      result
     }
-    None => (MediaType::from(path), None),
   }
-}
 
-fn map_js_like_extension(path: &Path, default: MediaType) -> MediaType {
-  match path.extension() {
-    None => default,
-    Some(os_str) => match os_str.to_str() {
-      None => default,
-      Some("jsx") => MediaType::JSX,
-      Some("tsx") => MediaType::TSX,
-      // Because DTS files do not have a separate media type, or a unique
-      // extension, we have to "guess" at those things that we consider that
-      // look like TypeScript, and end with `.d.ts` are DTS files.
-      Some("ts") => {
-        if default == MediaType::TypeScript {
-          match path.file_stem() {
-            None => default,
-            Some(os_str) => {
-              if let Some(file_stem) = os_str.to_str() {
-                if file_stem.ends_with(".d") {
-                  MediaType::Dts
-                } else {
-                  default
-                }
-              } else {
-                default
-              }
-            }
-          }
-        } else {
-          default
-        }
-      }
-      Some(_) => default,
-    },
+  /// Get a previously in memory cached file.
+  pub fn get_cache(&self, specifier: &ModuleSpecifier) -> Option<File> {
```

Comment:
I'd expect to get an instance of `cache` from this method :) maybe `get_cached`?

Reply:
I also changed `insert_cache` to `insert_cached` to be consistent.